### ISSUE TITLE
Convert KRunner to org.apache.cli

### ISSUE THREAD

**Issue by adamsmd** [kframework/k-legacy#180]
KRunner should be converted to use org.apache.cli like the rest of the project instead of joptsimple which it is currently using.

Once that conversion is done we should remove lib/java/jopt-simple-3.3.jar as we will no longer depend on it.

(Note that KRunner appears to be an internally used program, but it will still simplify maintenance to drop that dependency.)

**Comment by denis-bogdanas** [comment]
I totally agree with this proposal.

I also suggest considering a more high-level option parsing Framework, like JCommander:
http://beust.com/jcommander/#Overview
In addition ot CLI it supports mapping options to fields in a class. Could greatly improve maintenance.

**Comment by dwightguth** [comment]
Resolving duplicate of #244. I will look into JCommander when the time comes.
